pkg/skaffold/instrumentation/export.go

```diff
@@ -283,3 +299,160 @@ func errorMetrics(ctx context.Context, meter skaffoldMeter, m metric.Meter, labe
 		unknownCounter.Record(ctx, 1, labels...)
 	}
 }
+
+type TraceExporterConfig struct {
+	writer    io.Writer
+	keyFS     http.FileSystem
+	keyPath   string
+	jaegerURL string
+}
+
+type TraceExporterOption func(te *TraceExporterConfig)
+
+func WithWriter(w io.Writer) TraceExporterOption {
+	return func(teconf *TraceExporterConfig) {
+		teconf.writer = w
+	}
+}
+
+func WithKeyFS(keyFS http.FileSystem) TraceExporterOption {
+	return func(teconf *TraceExporterConfig) {
+		teconf.keyFS = keyFS
+	}
+}
+
+func WithJaegerURL(jaegerURL string) TraceExporterOption {
+	return func(teconf *TraceExporterConfig) {
+		teconf.jaegerURL = jaegerURL
+	}
+}
+
+func WithKeyPath(keyPath string) TraceExporterOption {
+	return func(teconf *TraceExporterConfig) {
+		teconf.keyPath = keyPath
+	}
+}
+
+func initTraceExporter(opts ...TraceExporterOption) (trace.TracerProvider, func(context.Context) error, error) {
+	teconf := TraceExporterConfig{
+		writer:    os.Stdout,
+		keyFS:     nil,
+		keyPath:   "/secret/keys.json",
+		jaegerURL: "http://localhost:14268/api/traces",
+	}
+
+	for _, opt := range opts {
+		opt(&teconf)
+	}
+
+	if _, ok := os.LookupEnv("SKAFFOLD_EXPORT_TO_STDOUT"); ok {
+		logrus.Debugf("using stdout trace exporter")
+		return initIOWriterTracer(teconf.writer)
+	}
```

Comment:
Do we need this if we have `SKAFFOLD_TRACE=stdout`?

Reply:
I have removed this now.  Metrics was using this variable `SKAFFOLD_EXPORT_TO_STDOUT` and it wasn't clear to me what the intent was so I duped the functionality here but agree the logic/usage is confusing atm.